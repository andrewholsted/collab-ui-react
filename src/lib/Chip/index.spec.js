import React from 'react';
import { shallow } from 'enzyme';
import { Chip } from '@collab-ui/react';

describe('<Chip />', () => {
  it('should match SnapShot', () => {
    const container = shallow(<Chip type="recording" />);

    expect(container).toMatchSnapshot();
  });

  it('should render the correct left content for a recording chip', () => {
    const container = shallow(<Chip type="recording"/>);
    expect(container.find('.cui-chip-left.recording').length).toEqual(1);
  });

  it('should render the correct content for a file chip', () => {
    const container = shallow(<Chip type="file" fileType="audio" fileDownloadLink="https://download-file-link"/>);
    expect(container.find('.cui-chip-left.file').length).toEqual(1);
    expect(container.find('.cui-file-download').prop('href')).toBe('https://download-file-link');
  });

  it('should accept a custom class', () => {
    const container = shallow(<Chip type="recording" className="custom-recording-class"/>);
    expect(container.find('.custom-recording-class').length).toEqual(1);
  });

  it('should render custom left content', () => {
    const leftContent = <div className="custom-left"/>;
    const container = shallow(<Chip leftContent={leftContent}/>);
    expect(container.find('.custom-left').length).toEqual(1);
  });

  it('should render custom right content', () => {
    const rightContent = <div className="custom-right"/>;
    const container = shallow(<Chip rightContent={rightContent}/>);
    expect(container.find('.custom-right').length).toEqual(1);
  });
});
