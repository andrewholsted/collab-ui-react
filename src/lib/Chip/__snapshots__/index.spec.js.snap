// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> should match SnapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chip
    type="recording"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="cui-chip-left recording"
        >
          <i
            className="icon icon-play-circle_32 white"
          />
        </div>,
        <div
          className="cui-chip-center"
        />,
        undefined,
      ],
      "className": "cui-chip",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="icon icon-play-circle_32 white"
          />,
          "className": "cui-chip-left recording",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon icon-play-circle_32 white",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
          ],
          "className": "cui-chip-center",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cui-chip-left recording"
          >
            <i
              className="icon icon-play-circle_32 white"
            />
          </div>,
          <div
            className="cui-chip-center"
          />,
          undefined,
        ],
        "className": "cui-chip",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="icon icon-play-circle_32 white"
            />,
            "className": "cui-chip-left recording",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon icon-play-circle_32 white",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
            ],
            "className": "cui-chip-center",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
